compiler/z/codegen/S390Peephole.cpp

```diff
@@ -1349,26 +1349,66 @@ TR_S390PostRAPeephole::ConditionalBranchReduction(TR::InstOpCode::Mnemonic branc
 
 
 /**
- * Catch the pattern where an CR/BRC can be converted
- *    CR R1,R2
- *    BRC  Mask, Lable
+ * Catch the pattern where an CLR/BRC can be converted
+ *    CLR R1, R2
+ *    BRC Mask, Lable
  * Can be replaced with
- *    CRJ  R1,R2,Lable
+ *    CLRJ R1, R2, Lable, Mask
  */
 bool
 TR_S390PostRAPeephole::CompareAndBranchReduction()
    {
-   bool performed = false;
+   if (!TR::Compiler->target.cpu.getSupportsArch(TR::CPU::z10))
+      return false;
 
-   TR::Instruction *current = _cursor;
-   TR::Instruction *next = _cursor->getNext();
-   TR::InstOpCode::Mnemonic curOpCode = current->getOpCodeValue();
-   TR::InstOpCode::Mnemonic nextOpCode = next->getOpCodeValue();
-   if ((curOpCode == TR::InstOpCode::CR || curOpCode == TR::InstOpCode::CGR || curOpCode == TR::InstOpCode::CGFR)
-      && (nextOpCode == TR::InstOpCode::BRC || nextOpCode == TR::InstOpCode::BRCL))
-      {
-      printf("Finding CR + BRC\n");
-      printf("method=%s\n", comp()->signature());
+   bool performed = false;
+   TR::Compilation *comp = _cg->comp();
```

Comment:
This variable is only used in a single place so there is no need for it to exist. Do you agree that we can fold it into the single use? Also there is an existing `comp()` function used throughput the peephole optimizations. Perhaps we can use this API for consistency?